### DISCUSSION TITLE
Config option to disable editing M2M linked item

### DISCUSSION THREAD

**Discussion by monbrey** [directus/directus#16115] [Feature Requests]
First off, loving Directus and how easy both the UI and the APIs are to use.

I've got M2M relationships set up between some of my collections, which I'll refer to generically as Collection A => Junction => Collection B. When adding/editing a record in Collection A, and opening one of the M2M relationship records to edit additional values of the Junction table, I'd like to be able to control the read-only state of items/fields in Collection B in this interface

The combined interface showing properties of both Junction and Collection B is useful, but not entirely intuitive as it may appear to the user that they're making edits that apply only to this particular relationship record - rather they would be directly modifying the Collection B record.

**Comment by connorwinston** [comment]
Can you clarify where you are editing Collection B from?
- Collection A Item View
- Junction Item View
- Collection B Item View

Typically the junction collection is hidden and you make your edits on CollA or CollB item pages respectively and you can edit the related items using the M2M relationship interface.

**Reply by monbrey** [reply]
I'm not quite sure of the correct names, but the workflow would be as follows:

1. Open the Content panel, Collection A, and edit one of its items (Collection A item view?)
2. Add a new item from Collection B to the M2M field.
3. Open the junction item from the M2M table/list in the Collection A item view. This opens a panel on the right.
4. Edit/fill the additional fields of the junction item.

While editing the fields of the junction item, this panel in Step 4 also displays and allows editing of the fields of the Collection B item. I'd like a config option to control if this should be possible, or if they should be read-only and only allow fields of the junction item to be edited in this view.

**Reply by bee8bit** [reply]
I agree. Especially when it is not allowed to create new items in Collection B but only to select existing items.
I would expect to edit the junction fields (and possibly select a different item in Collection B, but deleting+adding another junction would do for the moment).

Suggestion: Can we extend the option "Junction Fields Location" by a new entry "Show only Junction Fields"?

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
